pkg/image/apis/image/helper_test.go

```diff
@@ -1510,3 +1510,125 @@ func TestValidateRegistryURL(t *testing.T) {
 		}
 	}
 }
+
+func TestFollowTagReference(t *testing.T) {
+	tests := map[string]struct {
+		stream      *ImageStream
+		tag         string
+		expFinalTag string
+		expRef      *TagReference
+		expMultiple bool
+		expErr      error
+	}{
+		"follow tag reference": {
+			stream: &ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testis",
+				},
+				Spec: ImageStreamSpec{
+					Tags: map[string]TagReference{
+						"mytag":   {From: &kapi.ObjectReference{Kind: "ImageStreamTag", Name: "sometag"}},
+						"sometag": {From: &kapi.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:sometag"}},
+					},
+				},
+			},
+			tag:         "mytag",
+			expFinalTag: "sometag",
+			expRef: &TagReference{
+				From: &kapi.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:sometag"},
+			},
+			expMultiple: true,
+			expErr:      nil,
+		},
+		"follow tag reference with istag:mytag format": {
+			stream: &ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testis",
+				},
+				Spec: ImageStreamSpec{
+					Tags: map[string]TagReference{
+						"mytag":   {From: &kapi.ObjectReference{Kind: "ImageStreamTag", Name: "testis:sometag"}},
+						"sometag": {From: &kapi.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:sometag"}},
+					},
+				},
+			},
+			tag:         "mytag",
+			expFinalTag: "sometag",
+			expRef: &TagReference{
+				From: &kapi.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:sometag"},
+			},
+			expMultiple: true,
+			expErr:      nil,
+		},
+		"no tag reference error": {
+			stream: &ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testis",
+				},
+				Spec: ImageStreamSpec{
+					Tags: map[string]TagReference{
+						"mytag":    {From: &kapi.ObjectReference{Kind: "ImageStreamTag", Name: "correcttag"}},
+						"wrongtag": {From: &kapi.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:mytag"}},
+					},
+				},
+			},
+			tag:         "mytag",
+			expFinalTag: "correcttag",
+			expRef:      nil,
+			expMultiple: true,
+			expErr:      ErrNotFoundReference,
+		},
+		"crosss image tag reference error": {
+			stream: &ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testis",
+				},
+				Spec: ImageStreamSpec{
+					Tags: map[string]TagReference{
+						"mytag":           {From: &kapi.ObjectReference{Kind: "ImageStreamTag", Name: "another:sometag"}},
+						"another:sometag": {From: &kapi.ObjectReference{Kind: "DockerImage", Name: "repo.com/somens/someimage:mytag"}},
```

Comment:
"another:sometag" shouldn't be a valid tag name i wouldn't think (contains a colon).

Reply:
Thank you. So, I think the line should be just removed  (becasue `"mytag":           {From: &kapi.ObjectReference{Kind: "ImageStreamTag", Name: "another:sometag"}},` has the invalid reference). I hope I didn't misunderstand it.